x/muxer/muxer.go

```diff
@@ -423,8 +430,77 @@ func (m *muxer) Attach(ctx context.Context, req *rpc.PluginAttach) (*emptypb.Emp
 	return &emptypb.Empty{}, nil
 }
 
-func (m *muxer) GetMapping(context.Context, *rpc.GetMappingRequest) (*rpc.GetMappingResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "UNIMPLEMENTED")
+func (m *muxer) GetMapping(ctx context.Context, req *rpc.GetMappingRequest) (*rpc.GetMappingResponse, error) {
+	subs := make([]func() tuple[*rpc.GetMappingResponse, error], len(m.servers))
+	for i, s := range m.servers {
+		i, s := i, s
+		subs[i] = func() tuple[*rpc.GetMappingResponse, error] {
+			return newTuple(s.GetMapping(ctx, proto.Clone(req).(*rpc.GetMappingRequest)))
+		}
+	}
+	errs := new(multierror.Error)
+	results := [][]byte{}
+	var providerName string
+	for i, v := range asyncJoin(subs) {
+		if err := v.B; err != nil {
+			errs.Errors = append(errs.Errors, err)
+			continue
+		}
+		response := v.A
+		if len(response.Data) == 0 {
+			continue
+		}
+		if response.GetProvider() == "" {
+			errs = multierror.Append(errs,
+				m.Warnf(ctx, "GetMapping", "Missing provider name for subprovider %d", i))
```

Comment:
AFAIK It always has to be set so can start being strict.

Reply:
Great, I'll make it strict.